mits/config/config.go

```diff
@@ -0,0 +1,106 @@
+/*
+   Copyright 2020 SUSE
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"gopkg.in/yaml.v2"
+)
+
+// ConfigLoader wraps the Load method for loading a configuration file.
+type ConfigLoader struct {
+	openFile         func(name string) (io.ReadCloser, error)
+	newConfigDecoder func(r io.Reader) configDecoder
+}
+
+// NewYAMLConfigLoader constructs a new ConfigLoader for loading YAML files.
+func NewYAMLConfigLoader() *ConfigLoader {
+	return &ConfigLoader{
+		openFile: openFile,
+		newConfigDecoder: func(r io.Reader) configDecoder {
+			decoder := yaml.NewDecoder(r)
+			decoder.SetStrict(true)
+			return decoder
+		},
+	}
+}
+
+func openFile(name string) (io.ReadCloser, error) {
```

Comment:
None of this DI stuff is actually used (you don't have a test for it), so it shouldn't exist.  I know I mentioned hidden dependencies… but if you're not doing anything about it, we should completely do nothing about it (rather than do it halfway and make things complicated for no benefit).

Reply:
:)